FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpEncryptAndOrSign.kt

```diff
@@ -109,15 +116,22 @@ object PgpEncryptAndOrSign {
     pgpPublicKeyRingCollection: PGPPublicKeyRingCollection,
     pgpSecretKeyRingCollection: PGPSecretKeyRingCollection?,
     secretKeyRingProtector: SecretKeyRingProtector?,
-    doArmor: Boolean
+    doArmor: Boolean,
+    passphrase: Passphrase? = null
   ): EncryptionStream {
     val encOpt = EncryptionOptions().apply {
-      pgpPublicKeyRingCollection.forEach {
-        addRecipient(it)
+      if (passphrase != null) {
+        addPassphrase(passphrase)
+      } else {
+        pgpPublicKeyRingCollection.forEach {
+          addRecipient(it)
+        }
```

Comment:
This method should probably be more generic. If there is a password, add it. If there is public keys, add them. Don't make it exclusive either/or here, even if currently we use it exclusively.